docs/source/developer_documentation/development_tutorial/creating_a_boefje.md

```diff
@@ -0,0 +1,146 @@
+- `properties`: This contains a list of objects which each will show the KAT-alogus what inputs are requested from the user. This can range from requesting for an API-key to extra commands the boefje should run.
+  Inside the `boefje.json` file, we specified 2 environment variables that will be used by this boefje.
+  - `MESSAGE`: For this property we ask the user to send us a string which this boefje will use to create some raw data.
+  - `NUMBER`: For this property we ask the user to send us an integer between 1 and 9.
+- `required`: In here we need to give a list of the objects' names that the user has to provide to run our boefje. For this example, we will only require the user to give us the `MESSAGE` variable. We do this by adding `"MESSAGE"` to the `required` list.
+
+## `main.py`
+
+This is the file where the boefje's meowgic happens. This file has to contain a run method that accepts a dictionary and returns a `list[tuple[set, bytes | str]]`.
+This function will run whenever a new OOI gets created with one of the types mentioned in `consumes` inside `boefje.json`. :
+
+Here is the example we will be using:
+
+```python
+import json
+from os import getenv
+
+def run(boefje_meta: dict) -> list[tuple[set, bytes | str]]:
+    """Function that gets run to give a raw file for the normalizers to read from"""
+    address = boefje_meta["arguments"]["input"]["address"]
+    MESSAGE = getenv("MESSAGE", "ERROR")
+    NUMBER = getenv("NUMBER", "0")
+
+    # Check if NUMBER has been given, if it has not. Keep it at 0
+    try:
+        amount_of_cats = int(NUMBER)
+    except _:
+        amount_of_cats = 0
+
+    cats = "😺" * amount_of_cats
+    greeting = f"{MESSAGE}{cats}!!!"
+
+    raw = json.dumps({
+        "address": address,
+        "greeting": greeting
+    })
+
+
+    return [
+        (set(), raw)
+    ]
+```
+
+The most important part is the return value we send back:
+The final task of creating a boefje is specifying what DockerFile our boefje should use. We can do this inside the file located in `boefjes/Makefile`.
+Inside the `images` rule. We have to specify that we want to use `base.Dockerfile` since our boefje does not require any special packages to run. This is as simple as adding a single line. Here is what that would look like in our case:
+
+**BEFORE**
+
+```
+images:  # Build the images for the containerized boefjes
+	docker build -f ./boefjes/plugins/kat_nmap_tcp/boefje.Dockerfile -t openkat/nmap  .
+```
+
+**AFTER**
+
+```
+images:  # Build the images for the containerized boefjes
+	docker build -f ./boefjes/plugins/kat_nmap_tcp/boefje.Dockerfile -t openkat/nmap  .
+	docker build -f images/base.Dockerfile -t openkat/hello-katty --build-arg BOEFJE_PATH=./boefjes/plugins/kat_hello_katty .
+```
+
+This was the creation of our first boefje. If we run OpenKAT now we should be able to see this boefje sitting in the KAT-alogus. Let’s try it out!
```

Comment:
I think you miss the reference here to the next part of the tutorial on creating the model? It is not referenced in the other tutorial pages.